drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -410,7 +410,17 @@ NTSTATUS TCPConnect
     Status = TCPTranslateError(LibTCPConnect(Connection,
                                                 &connaddr,
                                                 RemotePort));
-
+    
+    if ((Status != STATUS_SUCCESS) && (Status != STATUS_PENDING)) { 
+    	TI_DbgPrint(DEBUG_TCP,
+                    ("no pcb from LibTCPConnect, clear ConnectRequest.\n")); 
+    	LockObject(Connection);
+    	RemoveTailList( &Connection->ConnectRequest); 
```

Comment:
Are we sure the tail entry is still our `Bucket`? Use `RemoveEntryList()`?

```suggestion
    	RemoveTailList(&Connection->ConnectRequest); 
```